src/main/java/org/commcare/util/EncryptionUtils.java

```diff
@@ -1,49 +1,100 @@
 package org.commcare.util;
 
+import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.nio.charset.Charset;
 import java.security.InvalidAlgorithmParameterException;
 import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.KeyFactory;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
 import java.security.NoSuchAlgorithmException;
+import java.security.Security;
+import java.security.UnrecoverableEntryException;
+import java.security.cert.CertificateException;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
 
 import javax.crypto.BadPaddingException;
 import javax.crypto.Cipher;
 import javax.crypto.IllegalBlockSizeException;
 import javax.crypto.NoSuchPaddingException;
-import javax.crypto.SecretKey;
 import javax.crypto.spec.GCMParameterSpec;
-import javax.crypto.spec.IvParameterSpec;
 import javax.crypto.spec.SecretKeySpec;
 
 public class EncryptionUtils {
+    private static KeyStore androidKeyStore;
+
+    public static final String USER_CREDENTIALS_KEY_ALIAS = "user-credentials-key-alias";
+
+    public static final String ANDROID_KEYSTORE_PROVIDER_NAME = "AndroidKeyStore";
```

Comment:
Core classes should not contain any platform specific implementation details. You can probably use the [platform classes ](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/utils/AndroidCommCarePlatform.java) to abstract the keystore name for respective platforms in their respective platform implementations.

Reply:
Done 5b66cc4ab32a6c9c1356f56007e484bea0cabe2e. I opted for a `static non-final field` to store the name of the platform keyStore.